### DISCUSSION TITLE
Notion blocks implementation, image upload through Dropbox and markdown parser

### DISCUSSION THREAD

**Discussion by ghost** [ramnes/notion-sdk-py#106] [Show and tell]
Hi there! First of all, thanks for this awesome package!

Since few examples are available, it was very hard for me to figure out how to utilize `notion-sdk-py`. The main struggle I faced was trying to create new blocks to a Notion page. Therefore, I have created a repository that implements most Notion blocks as Python dictionaries that can be used in methods such as `notion.blocks.children.append `. Maybe this was already implemented somewhere else, but I did not manage to find anything! It also supports appending blocks, creating `rich_text` objects as well as add `annotations` for text formatting.

Also, I wanted to upload images (or basically any file) to Notion. Since the API does not allow to upload files directly, I had to choose Dropbox API as a way to upload and generate raw shared links for the files. That is how I can use external links for any Notion block supporting external links such as `image`,  `video`, `embed`, `file`... and so on.

Finally, I'm developing in my sparse time a markdown parser. It is in its very early stages, but maybe someone can find it useful too!

All examples can be found in this [Notion-API-SDK](https://github.com/AlbertoMartinPerez/Notion_API_SDK) repository I've created. Do you think it could be relevant for anyone using `notion-sdk-py`?
_(Please note that I'm figuring out yet how to use GIT for issues, pull requests and so on! Should I have shared this in another way?)_

**Comment by ramnes** [comment]
Hey Alberto, thanks for the kind words!

As mentioned yesterday in [another discussion](https://github.com/ramnes/notion-sdk-py/discussions/105), I'm definitely interested in more examples ­— as long as they're short and not complete applications. :)

For a higher-level library, you might be interested in @jheddings' Notional: https://github.com/jheddings/notional

**Comment by tyehuoPHD** [comment]
@AlbertoMartinPerez  Hello, thanks for your contribution. But i cant find your repository any more. I hope to use some of your code if that's possible. haha

**Reply by ghost** [reply]
I made it public again 😄

**Reply by tyehuoPHD** [reply]
> I made it public again 😄

thanks man !
